File: myadmin-angular/projects/parcel-tracking-app/src/app/services/tracking.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {Observable} from 'rxjs';
import {PackageInfo} from '../info-objects';

@Injectable()
export class TrackingService {

  constructor(private http:HttpClient) {
  }

  trackPackageByStudent(studentNumber:number):Observable<PackageInfo> {
    return this.http.get<PackageInfo>(`/myadmin-tracking-services/services/rest/postaltrackingservice/track/${studentNumber}`);
  }

}

File: myadmin-service-api/src/main/java/za/ac/unisa/myadmin/tracking/services/TrackingService.java
package za.ac.unisa.myadmin.tracking.services;

import za.ac.unisa.myadmin.common.exceptions.OperationFailedException;
import za.ac.unisa.myadmin.tracking.services.dto.PackageInfo;

public interface TrackingService {

	/**
	 * Processes a request to get tracking info from parcel proxy
	 *
	 * @param studentNumber
	 *            The student number for tracking
	 * @return A tracking info for the requested student
	 * @throws OperationFailedException
	 */
	public PackageInfo trackPackageByStudent(Integer studentNumber) throws OperationFailedException;
}
